File: api-tester/src/main/java/com/revenuecat/apitester/java/PresentedOfferingContextAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.PresentedOfferingContext;

@SuppressWarnings({"unused"})
final class PresentedOfferingContextAPI {
    static void check(final PresentedOfferingContext presentedOfferingContext) {
        final String offeringIdentifier = presentedOfferingContext.getOfferingIdentifier();
        final String placementIdentifier = presentedOfferingContext.getPlacementIdentifier();
        final PresentedOfferingContext.TargetingContext targetingContext = presentedOfferingContext.getTargetingContext();
    }

    static void checkConstructor(final String offeringIdentifier) {
        final PresentedOfferingContext presentedOfferingContext = new PresentedOfferingContext(offeringIdentifier, null, null);
    }
}

final class TargetingContextAPI {
    static void check(final PresentedOfferingContext.TargetingContext targetingContext) {
        final int revision = targetingContext.getRevision();
        final String ruleId = targetingContext.getRuleId();
    }

    static void checkConstructor(final int revision, final String ruleId) {
        final PresentedOfferingContext.TargetingContext targetingContext = new PresentedOfferingContext.TargetingContext(revision, ruleId);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PresentedOfferingContextAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.PresentedOfferingContext

@Suppress("unused", "UNUSED_VARIABLE")
private class PresentedOfferingContextAPI {
    fun check(presentedOfferingContext: PresentedOfferingContext) {
        val offeringIdentifier: String = presentedOfferingContext.offeringIdentifier
        val placementIdentifier: String? = presentedOfferingContext.placementIdentifier
        val targetingContext: PresentedOfferingContext.TargetingContext? = presentedOfferingContext.targetingContext
    }

    fun checkConstructor(
        offeringId: String,
        placementId: String?,
        targetingContext: PresentedOfferingContext.TargetingContext?,
    ) {
        val poc1 = PresentedOfferingContext(offeringId)
        val poc2 = PresentedOfferingContext(offeringId, placementId)
        val poc3 = PresentedOfferingContext(offeringId, placementId, targetingContext)
    }
}

@Suppress("unused", "UNUSED_VARIABLE")
private class TargetingContextAPI {
    fun check(targetingContext: PresentedOfferingContext.TargetingContext) {
        val revision: Int = targetingContext.revision
        val ruleID: String = targetingContext.ruleId
    }

    fun checkConstructor(
        revision: Int,
        ruleId: String,
    ) {
        val targetingContext = PresentedOfferingContext.TargetingContext(revision, ruleId)
    }
}
